airflow-core/src/airflow/ui/tests/e2e/pages/HomePage.ts

```
@@ -0,0 +1,181 @@
+    this.poolSummarySection = page.locator('text="Pool Summary"').locator("..").locator("..");
+
+    // Historical Metrics section (recent runs)
+    this.historicalMetricsSection = page.getByText("History").first().locator("..").locator("..");
+    this.dagRunMetrics = page.getByRole("heading", { name: /dag run/i }).first();
+    this.taskInstanceMetrics = page.getByRole("heading", { name: /task instance/i }).first();
+  }
+
+  /**
+   * Check if health badges are visible
+   */
+  public async areHealthBadgesVisible(): Promise<boolean> {
+    try {
+      await this.metaDatabaseHealth.waitFor({ state: "visible", timeout: 10_000 });
+      await this.schedulerHealth.waitFor({ state: "visible", timeout: 5000 });
+
+      return true;
+    } catch {
+      return false;
+    }
+  }
+
+  /**
+   * Get Active DAGs count
+   */
+  public async getActiveDagsCount(): Promise<number> {
+    return this.getStatsCardCount(this.activeDagsCard);
+  }
+
+  /**
+   * Get Failed DAGs count
+   */
+  public async getFailedDagsCount(): Promise<number> {
+    return this.getStatsCardCount(this.failedDagsCard);
+  }
+
+  /**
+   * Get Running DAGs count
+   */
+  public async getRunningDagsCount(): Promise<number> {
+    return this.getStatsCardCount(this.runningDagsCard);
+  }
+
+  /**
+   * Check if DAG Import Errors are displayed (only visible when errors exist)
+   */
+  public async isDagImportErrorsVisible(): Promise<boolean> {
+    try {
+      return await this.dagImportErrorsCard.isVisible();
+    } catch {
+      return false;
+    }
+  }
+
+  /**
+   * Check if historical metrics (recent runs) section is visible
+   */
+  public async isHistoricalMetricsSectionVisible(): Promise<boolean> {
+    try {
+      await this.dagRunMetrics.waitFor({ state: "visible", timeout: 10_000 });
+
+      return true;
+    } catch {
+      return false;
+    }
+  }
+
+  /**
+   * Check if stats section is visible
+   */
+  public async isStatsSectionVisible(): Promise<boolean> {
```

Instead of creating own func. I think we should use Playwright assertions directly for clearer error messages.
This will fail silently

---

addressed